extensions/iceberg/build.gradle

```diff
@@ -23,6 +23,7 @@ dependencies {
 
     Classpaths.inheritIcebergCore(project)
     Classpaths.inheritIcebergHadoop(project)
+    Classpaths.inheritAvro(project, JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME)
```

Comment:
What was the specific error when this was not included?

It seems like we should be marking this as runtimeOnly instead? Maybe best to move into one of the added-back-in dependencies with a "because" reason in inheritParquetHadoopConfiguration.

Reply:
The error is:
```
/Users/larrybooker/dhc/lab-reduce-dep/extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergKeyValuePartitionedLayout.java:91: error: cannot access IndexedRecord
        final PartitionData partitionData = (PartitionData) df.partition();
                                            ^
  class file for org.apache.avro.generic.IndexedRecord not found
/Users/larrybooker/dhc/lab-reduce-dep/extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergKeyValuePartitionedLayout.java:94: error: cannot access SpecificData
            final Object colValue = partitionData.get(colData.index);
                                                 ^
  class file for org.apache.avro.specific.SpecificData not found
Note: /Users/larrybooker/dhc/lab-reduce-dep/extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
```

Reply:
Interesting; I wonder why iceberg doesn't declare this as an API dependency... I might file an issue with them.